modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/Treeview.tsx

```diff
@@ -848,23 +895,17 @@ Treeview.defaultProps = {
 	multiSelection: true,
 };
 
-Treeview.propTypes = {
-	NodeComponent: PropTypes.func,
-	filter: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-	inheritSelection: PropTypes.bool,
-	initialSelectedNodeIds: PropTypes.arrayOf(PropTypes.string),
-	multiSelection: PropTypes.bool,
-	nodes: PropTypes.arrayOf(
-		PropTypes.shape({
-			children: PropTypes.array,
-			expanded: PropTypes.bool,
-			id: PropTypes.string.isRequired,
-		})
-	).isRequired,
-	onSelectedNodesChange: PropTypes.func,
-};
-
 Treeview.Card = TreeviewCard;
 Treeview.Label = TreeviewLabel;
 
 export default Treeview;
+
+interface IProps {
+	NodeComponent: React.ComponentType<{node: Node}>;
+	filter?: NodeFilter | string;
+	inheritSelection: boolean;
+	initialSelectedNodeIds: string[];
+	multiSelection: boolean;
+	nodes: Node[];
+	onSelectedNodesChange: (selectedNodeIds?: Set<string>) => void;
```

Comment:
Using a `Set` here seems inconsistent with the previous prop type and also inconsistent with `initialSelectedNodeIds` which is `string[]`

Reply:
I think I would probably change all `Set<string>` to `string[]`

Reply:
Initially I've set all the types to `unknown` and them I've started follow the clues to to figure out which type was a better fit and the `Set<String>` was the perfect match. Change to `string[]` will force me to change the existent code to accommodate this type. 

I've assumed the `selectedNodeIds` is a `Set<string>` since we call `.has()` from `selectedNodeIds` in some points of the code:  
```tsx
function toggleNode(node: LinkedNode, selectedNodeIds: Set<String>) {
	if (node.selected !== selectedNodeIds.has(node.id)) {
``` 

```tsx
function computeParentSelection(
	nodeId: string | null,
	selectedNodeIds: Set<string>,
	nodes: NodeMap
): Set<string> {
	...
	const allChildrenSelected = node.children.every((children) =>
		selectedNodeIds.has(children.id)
	);
```

```tsx
function reducer(state: TreeviewState, action: TreeviewAction) {
	const {filteredNodes, nodeMap} = state;
	...
		case 'TOGGLE_SELECT': {
			....
					if (selectedNodeIds.has(id)) {

Reply:
I got the following errors  changing all the `Set<string>` to `string[]`:
```console
yarn run v1.22.17
$ liferay-npm-scripts build
Using pre-existing NODE_ENV: "development"
src/main/resources/META-INF/resources/treeview/Treeview.tsx(77,19): error TS2339: Property 'has' does not exist on type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(83,41): error TS2339: Property 'has' does not exist on type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(88,41): error TS2339: Property 'has' does not exist on type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(93,47): error TS2345: Argument of type 'string[] | Set<string>' is not assignable to parameter of type 'string[]'.
  Type 'Set<string>' is missing the following properties from type 'string[]': length, pop, push, concat, and 21 more.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(608,26): error TS2339: Property 'has' does not exist on type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(625,7): error TS2345: Argument of type 'Set<string>' is not assignable to parameter of type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(630,26): error TS2339: Property 'has' does not exist on type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(631,7): error TS2322: Type 'Set<string>' is not assignable to type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(638,7): error TS2322: Type 'Set<string>' is not assignable to type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(641,7): error TS2322: Type 'Set<string>' is not assignable to type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(650,25): error TS2345: Argument of type 'string[]' is not assignable to parameter of type 'Set<String>'.
  Type 'string[]' is missing the following properties from type 'Set<String>': add, clear, delete, has, and 2 more.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(656,35): error TS2345: Argument of type 'string[]' is not assignable to parameter of type 'Set<String>'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(688,44): error TS2339: Property 'has' does not exist on type 'string[]'.
src/main/resources/META-INF/resources/treeview/Treeview.tsx(786,28): error TS2769: No overload matches this call.
  Overload 1 of 3, '(reducer: Reducer<TreeviewState, TreeviewAction>, initializerArg: { filter?: string | NodeFilter | undefined; inheritSelection: boolean; initialNodes: Node[]; initialSelectedNodeIds: string[]; multiSelection: boolean; } & TreeviewState, initializer: (arg: { ...; } & TreeviewState) => TreeviewState): [...]', gave the following error.
    Type 'string | NodeFilter | undefined' is not assignable to type '((string | NodeFilter) & NodeFilter) | undefined'.
      Type 'string' is not assignable to type '((string | NodeFilter) & NodeFilter) | undefined'.
  Overload 2 of 3, '(reducer: Reducer<TreeviewState, TreeviewAction>, initializerArg: { filter?: string | NodeFilter | undefined; inheritSelection: boolean; initialNodes: Node[]; initialSelectedNodeIds: string[]; multiSelection: boolean; }, initializer: (arg: { ...; }) => TreeviewState): [...]', gave the following error.
    Argument of type '({ filter, inheritSelection, initialNodes, initialSelectedNodeIds, multiSelection, }: { filter?: string | NodeFilter | undefined; inheritSelection: boolean; initialNodes: Node[]; initialSelectedNodeIds: string[]; multiSelection?: boolean | undefined; }) => { ...; }' is not assignable to parameter of type '(arg: { filter?: string | NodeFilter | undefined; inheritSelection: boolean; initialNodes: Node[]; initialSelectedNodeIds: string[]; multiSelection: boolean; }) => TreeviewState'.
      Call signature return types '{ active: boolean; filter: NodeFilter | undefined; filteredNodes: LinkedNode[] | null; focusedNodeId: null; inheritSelection: boolean; multiSelection: boolean | undefined; nodeMap: {}; nodes: LinkedNode[]; selectedNodeIds: Set<...>; }' and 'TreeviewState' are incompatible.
        The types of 'selectedNodeIds' are incompatible between these types.
          Type 'Set<string>' is not assignable to type 'string[]'.
Command /home/mmcj/repository/liferay/liferay-portal/modules/node_modules/@liferay/npm-scripts/node_modules/.bin/tsc --emitDeclarationOnly exited with code 2
Received 1
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```